docs/debugging.md

```diff
@@ -1,239 +1,297 @@
 # Debug onos-config in Onit Using Delve
 
-## Debugging Example #1
-- In this tutorial, we assume you already started an onit cluster. If you haven't done that yet, refer to [testing](testing.md) for more details. 
+## Debugging Example #1: Use Goland IDE (Recommended):
+- In this example, we assume you already started an onit cluster. If you haven't done that yet, refer to [testing](testing.md) for more details.
+
+- First, you need to configure your [Goland](https://www.jetbrains.com/go/) IDE to connect to the remote [Delve](https://github.com/go-delve/delve/tree/master/Documentation/installation) debugger which is running in the onos-config 
+pod. You can do that by going to *Run | Edit Configurations… | + | Go Remote* and configuring the host (in this example *localhost*) and port (in this example port **40000**) your remote debugger is listening on.
+
+  ![goland_ide_configuration](images/goland_ide_remote_debug.gif)
+
+- To start debugging and  open a debugger port to the given resource, you can run `onit debug` command. For example, to forwards a port to the debug port on the onos-config pod, run the following command:
+  
+  ```bash
+  >onit debug <name of onos-config pod>
+  Forwarding from 127.0.0.1:40000 -> 40000
+  Forwarding from [::1]:40000 -> 40000
+  ```
+
+- To connect to the debugger using Goland IDE, select the configuration that you created 
+in the first step and click on *Debug* button as follows:
+
+   ![goland_ide_connect](images/goland_debug_connect.gif)
+
+
+- If your Goland IDE connects to the debug server which is running in onos-config pod successfully, then you should be able to see the following message in the logs:
+
+  ```bash
+   > onit get logs <name of onos-config pod>
+   ....
+   2019-07-22T15:09:27Z debug layer=debugger halting
+   2019-07-22T15:09:28Z debug layer=debugger continuing
+  ```
+
+- Suppose we want to add a simulator and debug [synchronizer](https://github.com/onosproject/onos-config/blob/master/pkg/southbound/synchronizer/synchronizer.go)
+code by setting a few breakpoints in different lines of its file as follows:
+
+  ![goland_ide_set_breakpoints](images/goland_debug_set_breakpoints.gif) 
+                 
+- As you can see above, after setting the breakpoints, you should be able to see some messages
+in the onos-config pod logs that confirm you set the breakpoints. When we add a simulator
+the Goland IDE points to the first breakpoint when is hit and you can go through the list 
```

Comment:
I'd reformulate with `the Goland IDE hits the first breakpoint that was set. At this point you can go....`